quarkchain/cluster/shard_db_operator.py

```diff
@@ -222,8 +224,9 @@ def skip_tx(normal_tx: EvmTransaction):
                 receipt = m_block.get_receipt(self.db, index)
                 tx = m_block.tx_list[index]  # type: TypedTransaction
                 evm_tx = tx.tx.to_evm_tx()
-                if not skip_tx(evm_tx):
+                if tx.get_hash() not in tx_hashes and not skip_tx(evm_tx):
```

Comment:
use a variable to store `get_hash()` result since it involves a serialization and SHA calculation